src/Microsoft.ML.TimeSeries/STL/InnerStl.cs

```diff
@@ -0,0 +1,321 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.TimeSeries
+{
+    internal class InnerStl
+    {
+        private readonly bool _isTemporal;
+        private double[] _seasonalComponent;
+        private double[] _trendComponent;
+        private double[] _residual;
+
+        // arrays for intermediate results
+        private List<double>[] _cycleSubSeries;
+        private List<double>[] _smoothedSubseries;
+
+        private double[] _s;
+        private double[] _t;
+        private double[] _detrendedY;
+        private double[] _c;
+        private double[] _deseasonSeries;
+
+        /// <summary>
+        /// the smoothing parameter for the seasonal component.
+        /// should be odd, and at least 7.
+        /// </summary>
+        private const int Ns = 9;
+
+        /// <summary>
+        /// the number of passes through the inner loop. /ref this value is set to 2, which works for many cases
```

Comment:
What is /ref ?

Reply:
Removed.

---
In reply to: [442524429](https://github.com/dotnet/machinelearning/pull/5202#discussion_r442524429) [](ancestors = 442524429)